code/modules/power/supermatter/supermatter.dm

```diff
@@ -497,24 +459,216 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
+	additive_power[SM_POWER_EXTERNAL_IMMEDIATE] = external_power_immediate
+	external_power_immediate = 0
+	additive_power[SM_POWER_HEAT] = gasmix_power_ratio * absorbed_gasmix.temperature / 6
+	
+	// I'm sorry for this, but we need to calculate power lost immediately after power gain.
+	// Helps us prevent cases when someone dumps superhothotgas into the SM and shoots the power to the moon for one tick.
+	/// Power if we dont have decay. Used for powerloss calc.
+	var/momentary_power = power
+	for(var/powergain_type in additive_power)
+		momentary_power += additive_power[powergain_type]
+	if(power < powerloss_linear_threshold) // Negative numbers
+		additive_power[SM_POWER_POWERLOSS] = -1 * (momentary_power / POWERLOSS_CUBIC_DIVISOR) ** 3 
+	else
+		additive_power[SM_POWER_POWERLOSS] = -1 * momentary_power * POWERLOSS_LINEAR_RATE + powerloss_linear_offset
+	// Positive number
+	additive_power[SM_POWER_POWERLOSS_GAS] = -1 * powerloss_dynamic_scaling *  additive_power[SM_POWER_POWERLOSS]
+	additive_power[SM_POWER_POWERLOSS_SOOTHED] = -1 * min(1-powerloss_dynamic_scaling , 0.2*psyCoeff) *  additive_power[SM_POWER_POWERLOSS]
+
+	for(var/powergain_types in additive_power)
+		power += additive_power[powergain_types]
+	power = max(power, 0)
+	return additive_power
+
+/**
+ * Perform calculation for the waste gas generated.
+ * Waste oxygen, plasma, and energy is just: constant * waste multiplier
+ * Waste multiplier is given by: power * constant * (1 + dynamic_heat_mod + psyCoeff)
+ * We just write it in a funny way to make it additive.
+ * 
+ * Updates:
+ * [/obj/machinery/power/supermatter_crystal/var/total_output_energy]
+ * 
+ * Returns: The factors that have influenced the calculation. list[FACTOR_DEFINE] = number
+ */
+/obj/machinery/power/supermatter_crystal/proc/calculate_waste_gas()
+	if(!produces_gas)
 		return
-	switch(type)
-		if(FLUX_ANOMALY)
-			var/explosive = has_changed_lifespan ? FLUX_NO_EXPLOSION : FLUX_LOW_EXPLOSIVE
-			new /obj/effect/anomaly/flux(local_turf, has_changed_lifespan ? rand(250, 350) : null, FALSE, explosive)
-		if(GRAVITATIONAL_ANOMALY)
-			new /obj/effect/anomaly/grav(local_turf, has_changed_lifespan ? rand(200, 300) : null, FALSE)
-		if(PYRO_ANOMALY)
-			new /obj/effect/anomaly/pyro(local_turf, has_changed_lifespan ? rand(150, 250) : null, FALSE)
-		if(HALLUCINATION_ANOMALY)
-			new /obj/effect/anomaly/hallucination(local_turf, has_changed_lifespan ? rand(150, 250) : null, FALSE)
-		if(VORTEX_ANOMALY)
-			new /obj/effect/anomaly/bhole(local_turf, 20, FALSE)
-		if(BIOSCRAMBLER_ANOMALY)
-			new /obj/effect/anomaly/bioscrambler(local_turf, null, FALSE)
+	/// Tell people the heat output in energy. More informative than telling them the heat multiplier.
+	var/additive_output_energy = list()
+	additive_output_energy[SM_WASTE_ENERGY] = INVERSE(12.5) * power ** 2
+	additive_output_energy[SM_WASTE_GAS] = dynamic_heat_modifier * INVERSE(12.5) * power ** 2
```

Comment:
Yep, squaring the heat penalties too, 86d9758. 

Squaring it inside this code block instead of there would mean that we are lying so much to the players with the UI and I cannot allow that. 

Downside is the negative multipliers get really small. Trying to "fix" that with a53bdb8. Still just testing the waters though, seeing what sticks. If i found something I'm happy with ill put it in the PR body.